examples/direct_fidelity_estimation.py

```diff
@@ -57,25 +56,31 @@ def main():
   # Computes for every \hat{P_i} of https://arxiv.org/pdf/1104.3835.pdf,
   # estimate rho_i and Pr(i). We then collect tuples (rho_i, Pr(i), \hat{Pi})
   # inside the variable 'pauli_traces'.
-  pauli_traces = [
-      compute_characteristic_function(circuit, P_i, qubits, noise=None) + (P_i,)
-      for P_i in itertools.product([cirq.I, cirq.X, cirq.Y, cirq.Z],
-                                           repeat=n)
-  ]
-  highest_probs = heapq.nlargest(n, pauli_traces, key=lambda e: e[1])
+  pauli_traces = []
+  for P_i in itertools.product([cirq.I, cirq.X, cirq.Y, cirq.Z], repeat=n):
+    rho_i, Pr_i = compute_characteristic_function(
+        circuit, P_i, qubits, noise=None)
+    pauli_traces.append({'P_i': P_i, 'rho_i': rho_i, 'Pr_i': Pr_i})
+
+  assert len(pauli_traces) == 4**n
+
+  p = [x['Pr_i'] for x in pauli_traces]
+  assert numpy.isclose(sum(p), 1.0, 1e-6)
 
   fidelity = 0.0
-  N = 0
-  for rho_i, Pr_i, P_i in highest_probs:
-    if Pr_i == 0:
-      break
+  for iter in range(n):
```

Comment:
don't use the variable name `iter`. It shadows a python builtin

Reply:
Good point. Changed to _.